javascripts/discourse/initializers/upload-consent-initializer.js

```diff
@@ -11,19 +12,25 @@ function initialize(api) {
     .split("|")
     .map((id) => parseInt(id, 10));
 
-  api.composerBeforeSave(async () => {
+  api.composerBeforeSave(() => {
     const categoryId = composerService.model?.categoryId;
     const reply = composerService.model?.reply;
 
     if (!categoryId || !reply) {
-      return;
+      return Promise.resolve();
     }
 
     if (!enabledCategories.includes(categoryId) || !uploadRegexp.test(reply)) {
-      return;
+      return Promise.resolve();
     }
 
-    await modal.show(UploadConsent);
+    return new Promise((resolve) => {
+      modal.show(UploadConsent, {
+        model: {
+          savePost: resolve,
+        },
+      });
+    });
```

Comment:
After we add the change in core this function can still be `async` and this part will look like this:

```suggestion
    let canSave = false;
    await modal.show(UploadConsent, {
      model: {
        savePost: () => canSave = true;
      },
    });

    return canSave;
```